src/models/ConstrainedMetaModel.ts

```diff
@@ -55,6 +55,9 @@ export class ConstrainedTupleModel extends ConstrainedMetaModel {
     ).map((tupleModel) => {
       return tupleModel.value = tupleModel.value as ConstrainedReferenceModel;
     });
+    //Ensure no duplicate references
+    dependencyModels = [...new Set(dependencyModels)];
```

Comment:
You can create some helper function to do this, but it's not mandatory.

Reply:
That's true but yea, just going to keep it as is 🙂